packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -184,6 +184,19 @@ const TransferPanel = (): JSX.Element => {
           const { decimals } = selectedToken
           const amountRaw = utils.parseUnits(amount, decimals)
 
+          // check that a registration is not currently in progress
+          const l2RoutedAddress = (
+            await bridge.l2Bridge.l2GatewayRouter.functions.calculateL2TokenAddress(
+              selectedToken.address
+            )
+          )[0]
+
+          if (selectedToken.l2Address !== l2RoutedAddress) {
```

Comment:
Paranoid, but could add a toLowerCase() to both addresses in case something un-checksummed ends up on a list (can't hurt)